app/upgrades/v15/upgrades.go

```diff
@@ -24,20 +45,80 @@ func CreateV15UpgradeHandler(
 
+		// TODO: Our mint module needs to be migrated to v47 for minttypes.ModuleName
+		// https://github.com/cosmos/cosmos-sdk/pull/12363/files
+		// Set param key table for params module migration
+		for _, subspace := range keepers.ParamsKeeper.GetSubspaces() {
+			subspace := subspace
+
+			var keyTable paramstypes.KeyTable
+			switch subspace.Name() {
+			case authtypes.ModuleName:
+				keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+			case banktypes.ModuleName:
+				keyTable = banktypes.ParamKeyTable() //nolint:staticcheck
+			case stakingtypes.ModuleName:
+				keyTable = stakingtypes.ParamKeyTable() //nolint:staticcheck
+			// TODO: mint module v47?
+			// case minttypes.ModuleName:
+			// 	keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+			case distrtypes.ModuleName:
+				keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck
+			case slashingtypes.ModuleName:
+				keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+			case govtypes.ModuleName:
+				keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+			case crisistypes.ModuleName:
+				keyTable = crisistypes.ParamKeyTable() //nolint:staticcheck
+
+			// ibc types
+			case ibctransfertypes.ModuleName:
+				keyTable = ibctransfertypes.ParamKeyTable()
+			case icahosttypes.SubModuleName:
+				keyTable = icahosttypes.ParamKeyTable()
+			case icacontrollertypes.SubModuleName:
+				keyTable = icacontrollertypes.ParamKeyTable()
+			// wasm
+			case wasmtypes.ModuleName:
+				keyTable = wasmtypes.ParamKeyTable() //nolint:staticcheck
+			}
+
+			if !subspace.HasKeyTable() {
+				subspace.WithKeyTable(keyTable)
+			}
+		}
+
+		// Migrate Tendermint consensus parameters from x/params module to a deprecated x/consensus module.
+		// The old params module is required to still be imported in your app.go in order to handle this migration.
+		baseAppLegacySS := keepers.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+		baseapp.MigrateParams(ctx, baseAppLegacySS, &keepers.ConsensusParamsKeeper)
+
 		// Run migrations
 		logger.Info(fmt.Sprintf("pre migrate version map: %v", vm))
 		versionMap, err := mm.RunMigrations(ctx, cfg, vm)
+		if err != nil {
+			return nil, err
+		}
 		logger.Info(fmt.Sprintf("post migrate version map: %v", versionMap))
 
-		// x/Mint
-		// Double blocks per year (from 6 seconds to 3 = 2x blocks per year)
-		mintParams := keepers.MintKeeper.GetParams(ctx)
-		mintParams.BlocksPerYear *= 2
-		keepers.MintKeeper.SetParams(ctx, mintParams)
-		logger.Info(fmt.Sprintf("updated minted blocks per year logic to %v", mintParams))
+		// Anything to do with ConsensusParamsKeeper?
+
+		// Interchain Queries
+		icqParams := icqtypes.NewParams(true, nil)
+		keepers.ICQKeeper.SetParams(ctx, icqParams)
+
+		// update gov params to use a 20% initial deposit ratio, allowing us to remote the ante handler
+		// TODO: Add test for this
+		govParams := keepers.GovKeeper.GetParams(ctx)
```

Comment:
Note to self: do local upgrade then export genesis. Export mainnet genesis & diff module param changes - ensure things match up as expected